src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.ts

```diff
@@ -175,7 +175,7 @@ export class AdminEntityFieldComponent implements OnInit {
     this.schemaFieldsForm.valueChanges.subscribe((v) => {
       if (JSON.stringify(v) === JSON.stringify(this.data.entitySchemaField))
         return;
-      this.data.entitySchemaField = { ...this.getUpdatedSchemaField(v) };
+      Object.assign(this.data.entitySchemaField, this.getUpdatedSchemaField(v));
```

Comment:
We added this originally because we want to follow the pattern that a component doesn't create side effects (e.g. change an object that is passed in) and only makes changes by emitting Output properties. That makes undo/cancel operations easier and gives us clearer control and less issues over change detection.

One alternative that might also help fix the losing-focus bug: Can we change the form control to [updateOn](https://v17.angular.io/api/forms/AbstractControlOptions#updateOn) "blur" - and would that avoid repeated updates even if we keep the re-creating a new object here?

Reply:
Thanks, the blur seems to be working as expected (hadn’t used it before, so new thing for me)

I applied the blur event to the entire schemaFieldsForm, but during testing, I found that for new fields, the auto-generate ID function wasn’t getting triggered until we left the input. So, now added it specifically for default values.